pkg/controller/controllerutil/controllerutil.go

```diff
@@ -274,6 +275,11 @@ const ( // They should complete the sentence "Deployment default/foo has been ..
 // Note: changes made by MutateFn to any sub-resource (status...), will be
 // discarded.
 func CreateOrUpdate(ctx context.Context, c client.Client, obj client.Object, f MutateFn) (OperationResult, error) {
+	return CreateOrUpdateWithEqualities(ctx, c, obj, equality.Semantic, f)
+}
+
+// CreateOrUpdateWithEqualities is like CreateOrUpdate but allows customizing the equality check.
+func CreateOrUpdateWithEqualities(ctx context.Context, c client.Client, obj client.Object, eq conversion.Equalities, f MutateFn) (OperationResult, error) {
```

Comment:
Alternatively maybe add a variadic `opts` arg to `CreateOrUpdate` that allows setting a custom comparison?

Also, wouldn't it be better to use a function type for this rather than a single-method interface, given that functions can be implemented inline and interfaces can not?